src/session_pool/session.js

```diff
@@ -298,10 +301,11 @@ export class Session {
      * @private
      */
     _puppeteerCookieToTough(puppeteerCookie) {
+        const expiresDate = puppeteerCookie.expires ? puppeteerCookie.expires : this._calculateNewDate();
         return new Cookie({
             key: puppeteerCookie.name,
             value: puppeteerCookie.value,
-            expires: new Date(puppeteerCookie.expires),
+            expires: expiresDate,
```

Comment:
We should validate this a bit better because apparently `new Date()` is different from `new Date(undefined)` and from `new Date(null)` which is the same as `new Date(0)`...

```js
const isExpiresValid = puppeteerCookie.expires && typeof puppeteerCookie.expires === 'number'
const expires = expiresIsValid ? puppeteerCookie.expires : _getDefaultCookieExpirationDate()
return {
// ...
    expires, // <--- instead of expires: expiresDate
// ...
}
```

Reply:
The logic of cookie date validation has been changed.